clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusV2ClusterCachingAgent.java

```diff
@@ -0,0 +1,634 @@
+      return null;
+    }
+
+    if (region.getName() != data.get("region")) {
+      return null;
+    }
+
+    Job job = metricsSupport.readData( () -> {
+      try {
+        return titusClient.findJobByName(data.get("serverGroupName").toString());
+      } catch (io.grpc.StatusRuntimeException e) {
+        if (e.getStatus() == Status.NOT_FOUND) {
+          return null;
+        } else {
+          log.error("Failed to load job {} from titus {}:{}", data.get("serverGroupName").toString(), account.getName(), region.getName(), e);
+        }
+      }
+      return null;
+    });
+
+    OnDemandResult onDemandResult = minimalOnDemand(providerCache, job, data);
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "handleOnDemand"))
+      .record(System.currentTimeMillis() - startTime, MILLISECONDS);
+
+    return onDemandResult;
+  }
+
+  /**
+   * Avoid writing cache results to both ON_DEMAND and SERVER_GROUPS, etc.
+   *
+   * By writing a minimal record to ON_DEMAND only, we eliminate significant overhead (redis and network) at the cost
+   * of an increase in time before a change becomes visible in the UI.
+   *
+   * A change will not be visible until a caching cycle has completed.
+   */
+  private OnDemandResult minimalOnDemand(ProviderCache providerCache, Job job, Map<String, ?> data) {
+    String serverGroupKey = Keys.getServerGroupV2Key(job.getName(), account.getName(), region.getName());
+    Map<String, Collection<CacheData>> cacheResults = new HashMap<>();
+    if (job == null) {
+      // avoid writing an empty onDemand cache record (instead delete any that may have previously existed)
+      providerCache.evictDeletedItems(ON_DEMAND.ns, Collections.singletonList(serverGroupKey));
+    } else {
+      Map <String, Object> attributes = new HashMap<>();
+      attributes.put("cacheTime", new Date());
+      attributes.put("cacheResults", Collections.emptyMap());
+
+      CacheData cacheData = metricsSupport.onDemandStore( () -> new DefaultCacheData(
+        serverGroupKey,
+        10 * 60, // ttl is 10 minutes
+        attributes,
+        Collections.emptyMap()
+      ));
+
+      cacheResults.put(ON_DEMAND.ns, Collections.singletonList(cacheData));
+    }
+    Map<String, Collection<String>> evictions = job == null
+      ? Collections.emptyMap()
+      : Collections.singletonMap(SERVER_GROUPS.ns, Collections.singletonList(serverGroupKey));
+
+    log.info("minimal onDemand cache refresh (data: {}, evictions: {})", data, evictions);
+    return new OnDemandResult(getOnDemandAgentType(), new DefaultCacheResult(cacheResults), evictions);
+  }
+
+  @Override
+  public CacheResult loadData(ProviderCache providerCache) {
+    Long startTime = System.currentTimeMillis();
+
+    log.info("Describing items in {}", getAgentType());
+    List<CacheData> evictFromOnDemand = new ArrayList<>();
+    List<CacheData> keepInOnDemand = new ArrayList<>();
+
+
+    List<Job> jobs = titusClient.getAllJobsWithoutTasks();
```

Comment:
This is the meaty call to Titus. Running locally against us-east-1 this takes ~15s.